specifyweb/frontend/js_src/lib/components/Errors/JSONError.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { jsonStringify } from '../../utils/utils';
+import { className } from '../Atoms/className';
+import { TableIcon } from '../Molecules/TableIcon';
+
+type JSONResponse = {
+  exception: string;
+  message: string;
+  data: any;
+  formattedData: string;
+  traceback: string;
+};
+
+function createJsonResponse(error: string): JSONResponse {
+  const json = JSON.parse(error);
+  const jsonResponse = {
+    exception: json.exception,
+    message: json.message,
+    data: json.data,
+    formattedData: jsonStringify(json.data, 2),
+    traceback: json.traceback,
+  };
+  return jsonResponse;
+}
+
+export function formatJSONBackendResponse(error: string): JSX.Element {
+  const response = createJsonResponse(error);
+  if (response.exception == 'BusinessRuleException')
+    return formatBusinessRuleException(error);
+  else if (response.exception == 'TreeBusinessRuleException')
+    return formatTreeBusinessRuleException(error);
+  else return formatBasicResponse(error);
+}
+
+function JSONBackendResponseFooter({

```

Comment:
You could also configure ESLint to report naming convention errors for you - https://github.com/specify/specify7/tree/production/specifyweb/frontend/js_src#plugins